File: gr/clustered-sines.hs
-- clustered sines (jmcc) #2

import Sound.SC3.ID {- hsc3 -}
import Sound.SC3.Lang.Control.OverlapTexture {- hsc3-lang -}

-- > synthstat cs
cs :: UGen
cs =
    let n = 80
        f1 = rand 'α' 100 1100
        f2 = 4 * f1
        sp = let y = uclone' 'α' n (f1 + rand 'α' 0 f2)
             in klangSpec y (map (f1 /) y) (replicate n 0)
    in uclone 'α' 2 (klang AR 1 0 sp * (0.3 / fromIntegral n))

main :: IO ()
main = xfadeTextureU (4,4,maxBound) cs

File: gr/clustered-sines.fs
\ clustered sines (jmcc) #2

: cs
    80 { n }
    100 1100 Rand.ir { f1 }
    4 f1 *  { f2 }
    n 0 do 0 f2 Rand.ir f1 + { y } y f1 y / 0 loop n 3 * mce { param }
    1 0 param Klang.ar 0.3 n / * 2 clone
;
cs 4 4 inf xfade-texture
